agent/csi/plugin.go

```diff
@@ -74,6 +77,55 @@ func NewNodePlugin(name string, nodeID string) *NodePlugin {
 	}
 }
 
+// connect is a private method that sets up the identity client and node
+// client from a grpc client. it exists separately so that testing code can
+// substitute in fake clients without a grpc connection
+func (np *NodePlugin) connect(ctx context.Context, cc *grpc.ClientConn) error {
+	np.cc = cc
+	// first, probe the plugin, to ensure that it exists and is ready to go
+	idc := csi.NewIdentityClient(cc)
+	np.idClient = idc
+
+	np.nodeClient = csi.NewNodeClient(cc)
+
+	return nil
+}
+
+func (np *NodePlugin) Client() csi.NodeClient {
+	return np.nodeClient
+}
+
+// TODO(ameyag): return error
+func (np *NodePlugin) init(ctx context.Context) error {
+	probe, err := np.idClient.Probe(ctx, &csi.ProbeRequest{})
+	if err != nil {
+		return err
+	}
+	if probe.Ready != nil && !probe.Ready.Value {
+		// TODO(ameyag): retry?
+		return nil
```

Comment:
Return an error here, something along the lines of "plugin not ready". It doesn't matter yet, but it does need to be an error.